eskip/eskip.go

```diff
@@ -185,6 +179,85 @@ type RouteInfo struct {
 	ParseError error
 }
 
+// HasPredicate checks if the route has predicate with the given name.
+func (r *Route) HasPredicate(name string) bool {
+	for _, p := range r.Predicates {
+		if p.Name == name {
+			return true
+		}
+	}
+
+	return false
+}
+
+// GetPredicates returns all predicates that match the given name.
+func (r *Route) GetPredicates(name string) []*Predicate {
+	pp := make([]*Predicate, 0)
+	for _, p := range r.Predicates {
+		if p.Name == name {
+			pp = append(pp, p)
+		}
+	}
+
+	return pp
+}
+
+// GetPredicate returns matching predicate or error when multiple predicates are given.
+func (r *Route) GetPredicate(name string) (*Predicate, error) {
+	pp := make([]*Predicate, 0)
+	for _, p := range r.Predicates {
+		if p.Name == name {
+			pp = append(pp, p)
+		}
+	}
+
+	if len(pp) == 0 {
+		return nil, nil
+	}
+
+	if len(pp) > 1 {
+		return nil, fmt.Errorf("multiple predicates of the same name: %d %s", len(pp), name)
+	}
+
+	return pp[0], nil
+}
+
+func (r *Route) PrependPredicate(pp ...*Predicate) error {
```

Comment:
Performance nitpicking: this would be possible to do by just appending once to a temp var, checking the temp var, and then if it is fine, setting the reference on the route to the var. This way there is only one allocation. Pseudo:

```
next := append(pp, r.Predicates...)
if len(PredicatesByName(next, "Path")) > 1 || len(PredicatesByName(next, "Weight")) > 1 {
    return fmt.Errorf()
}

r.Predicates = next
return nil
```

Reply:
please also check for "PathSubtree"

Reply:
This PR doesn't touch PathSubtree (yet) so that will be done in another PR.